app/policies/business_objective_policy.rb

```diff
@@ -3,20 +3,17 @@
 # Policies for accessing BusinessObjectives
 class BusinessObjectivePolicy < ApplicationPolicy
   def index?
-    record.profiles.pluck(:account_id).include? user.account_id
+    record.in_account(user.account_id)
   end
 
   def show?
-    record.profiles.pluck(:account_id).include? user.account_id
```

Comment:
This is the kind of thing we won't notice in a dev/CI environment where we only have a handful of accounts.